libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/NotificationToEmailConverterCommon.java

```diff
@@ -0,0 +1,152 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.yahoo.athenz.common.server.notification;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.text.MessageFormat;
+import java.util.Map;
+import java.util.ResourceBundle;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class NotificationToEmailConverterCommon {
+    private static final Logger LOGGER = LoggerFactory.getLogger(NotificationToEmailConverterCommon.class);
+
+    private static final String AT = "@";
+    private static final String USER_DOMAIN_DEFAULT = "user";
+    private static final String PROP_USER_DOMAIN = "athenz.user_domain";
+    private static final String PROP_NOTIFICATION_EMAIL_DOMAIN_TO = "athenz.notification_email_domain_to";
+    private static final String PROP_NOTIFICATION_WORKFLOW_URL = "athenz.notification_workflow_url";
+    private static final String PROP_NOTIFICATION_ATHENZ_UI_URL = "athenz.notification_athenz_ui_url";
+    private static final String PROP_NOTIFICATION_SUPPORT_TEXT = "athenz.notification_support_text";
+    private static final String PROP_NOTIFICATION_SUPPORT_URL = "athenz.notification_support_url";
+    private static final String EMAIL_TEMPLATE_CSS = "emails/base.css";
+
+    private static final String HTML_STYLE_TAG_START = "<style>";
+    private static final String HTML_STYLE_TAG_END = "</style>";
+    private static final String HTML_TBODY_TAG_START = "<tbody>";
+    private static final String HTML_TBODY_TAG_END = "</tbody>";
+
+    // can be moved to constructor which can take Locale as input parameter and return appropriate resource bundle
+    private static final ResourceBundle RB = ResourceBundle.getBundle("messages/ServerCommon");
+
+    private String userDomainPrefix;
+    private String emailDomainTo;
+    private String workflowUrl;
+    private String athenzUIUrl;
+    private String supportText;
+    private String supportUrl;
+    private String emailBaseCSS;
+
+
+    public NotificationToEmailConverterCommon() {
+        String userDomain = System.getProperty(PROP_USER_DOMAIN, USER_DOMAIN_DEFAULT);
+        userDomainPrefix = userDomain + "\\.";
+        emailDomainTo = System.getProperty(PROP_NOTIFICATION_EMAIL_DOMAIN_TO);
+        workflowUrl = System.getProperty(PROP_NOTIFICATION_WORKFLOW_URL);
+        athenzUIUrl = System.getProperty(PROP_NOTIFICATION_ATHENZ_UI_URL);
+        supportText = System.getProperty(PROP_NOTIFICATION_SUPPORT_TEXT);
+        supportUrl = System.getProperty(PROP_NOTIFICATION_SUPPORT_URL);
+
+        emailBaseCSS = readContentFromFile(getClass().getClassLoader(), EMAIL_TEMPLATE_CSS);
+
+    }
+
+    public String readContentFromFile(ClassLoader classLoader, String fileName) {
```

Comment:
This method gets ClassLoader as it will be called from both ZTS and ZMS (each with its own resources).